src/util.h

```diff
@@ -300,58 +301,81 @@ class MaybeStackBuffer {
     return buf_[index];
   }
 
+  // Size of the buffer
   size_t length() const {
     return length_;
   }
 
-  // Call to make sure enough space for `storage` entries is available.
-  // There can only be 1 call to AllocateSufficientStorage or Invalidate
-  // per instance.
+  // Actual maximum capacity of the buffer with which SetLength() can be called
+  // without first calling AllocateSufficientStorage()
+  size_t capacity() const {
+    return IsAllocated() ? capacity_ :
+                           IsInvalidated() ? 0 : kStackStorageSize;
+  }
+
+  // Make sure enough space for `storage` entries is available.
+  // This method can be called multiple times throughout the lifetime of the
+  // buffer, but once this has been called Invalidate() cannot be used.
+  // Contents of the buffer in the range [0, length()) are preserved.
```

Comment:
s/are/is/?  'Contents' is indefinite and therefore singular (I think?  Not 100% sure.)